src/main/java/games/strategy/triplea/delegate/MustFightBattle.java

```diff
@@ -480,8 +480,7 @@ private void removeAirNoLongerInTerritory() {
     // remove any air units that were once in this attack, but have now
     // moved out of the territory
     // this is an ilegant way to handle this bug
-    final CompositeMatch<Unit> airNotInTerritory = new CompositeMatchAnd<>();
-    airNotInTerritory.add(Matches.unitIsInTerritory(m_battleSite).invert());
+    final Match<Unit> airNotInTerritory = Matches.unitIsInTerritory(m_battleSite).invert();
```

Comment:
There was only one component match passed to the `CompositeMatchAnd` constructor here, so I didn't even bother to replace it with a `Match#all()` call.

I'm not sure, but there may be a bug here.  It looks like this removes all units not in the territory, not just air units.  Although, this could have been an intentional change, and the author simply didn't bother to rename the method.  I'll see if I can track it down in the history.